routers/web/repo/repo.go

```diff
@@ -347,10 +347,9 @@ func Action(ctx *context.Context) {
 		err = repo_model.WatchRepo(ctx, ctx.Doer, ctx.Repo.Repository, true)
 	case "unwatch":
 		err = repo_model.WatchRepo(ctx, ctx.Doer, ctx.Repo.Repository, false)
-	case "star":
-		err = repo_model.StarRepo(ctx, ctx.Doer, ctx.Repo.Repository, true)
-	case "unstar":
-		err = repo_model.StarRepo(ctx, ctx.Doer, ctx.Repo.Repository, false)
+	case "star", "unstar":
+		err = repo_model.StarRepo(ctx, ctx.Doer,
+			ctx.Repo.Repository, ctx.PathParam("action") == "star")
```

Comment:
```suggestion
	case "star":
		err = repo_model.StarRepo(ctx, ctx.Doer, ctx.Repo.Repository, true)
	case "unstar":
		err = repo_model.StarRepo(ctx, ctx.Doer, ctx.Repo.Repository, false)
```

revert unnecessary changes